### DISCUSSION TITLE
Welcome to zpm Discussions!

### DISCUSSION THREAD

**Discussion by isc-rsingh** [intersystems/ipm#321] [Announcements]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by isc-rsingh** [comment]
I'll start off. Hi. I'm a product manager at InterSystems working on the IRIS data platform. I'm managing the effort to allow the package manager to be used for the core product as well as for 3rd-party developers.

**Comment by mdavidovichHst** [comment]


**Reply by isc-tleavitt** [reply]
@mdavidovichHst I think you may have meant for this to be a separate discussion @ https://github.com/intersystems/ipm/discussions rather than under this thread.

IPM is a great tool for "I have some IRIS code on the filesystem and want to load it into the database, maybe with some additional configuration steps" and that's its main use in those sample repos.

**Reply by evshvarov** [reply]
Hi Mike @mdavidovichHst - we can take it offline and talk about IPM in community templates. But long story short - I think for 99% of projects that consists of InterSystems code elements it is make a lot of sense to follow IPM-package first approach. Even if you don't plan to use you own IPM registries. There are a lot of advantages starting from automatic tests and git-source control for interoperability UI elements and more.

**Reply by mdavidovichHst** [reply]
@isc-tleavitt yes, you're right - I don't 'forum' enough to recognize these conventions.

**Reply by daimor** [reply]
If you would develop some project on NodeJS, you most probably would use npm, in Java it would be mvn or graddle. So, ipm is like those tools, just helps build project, even small ones. And if you would wish to publish it, you already have mostly everything ready. And another thing, is to have your code organized, no matter the project you joined in which company, you may easily recognize the pattern, and understand the code structure

**Comment by Bitnerd79** [comment]
Bitnerd79, because we are all in the business of hauling bits. Work for MediQuant who recently acquired Knowledge Based Systems and their product KB_SQL. KB_SQL allows you to map legacy MUMPS globals to access them with SQL. KB_SQL also provides drivers for other applications to connect to KB_SQL using ODBC, JDBC, and ADO.NET.
